portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,34 +26,47 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private Map<String, String> comments = new HashMap<String, String>();
-
-    @Override
-    public void init() {
-        comments.put("Mario", "Hi!");
-        comments.put("Mario's imaginary friend", "Hi! What are you going?");
-        comments.put("Mario", "Just learning how to use JSON");
-        comments.entrySet().forEach(entry->{
-            System.out.println(entry.getKey() + ": " + entry.getValue());  
-        });
-    } 
-
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String json = convertToJson(comments);
-        System.out.println("Sending string: " + json);
-        response.setContentType("application/json;");
-        response.getWriter().println(json);
-    }
+  // IMPROVE: redundently storing display names
+  private List<List<String>> comments = new ArrayList<List<String>>();
+
+  @Override
+  public void init() {
+    comments.add(new ArrayList<String>());
```

Comment:
I am not fully sure why the data is organized this way. Maybe add a comment where you declared the variable, saying what each row is storing.

Can you add a method getCommentRow(comments, key, value) 
then init can just be
comments.add(getCommentRow("Mario", "Hi") )
comments.add(getCommentRow(-, -) )
comments.add(getCommentRow(-,-) )

List<String> getCommentRow(key, value) {
  List<String> row = new ArrayList...
  row.add()
  return row;
}

Reply:
Done: d22eb99